src/components/sound.js

```diff
@@ -155,7 +156,7 @@ module.exports.Component = registerComponent('sound', {
     this.pool.children.forEach(function (sound) {
       sound.onEnded = function () {
         sound.isPlaying = false;
-        el.emit('sound-ended', {index: i});
+        el.emit('sound-ended', {index: i, id: self.id, name: self.name});
```

Comment:
I believe that the index sent here is actually the wrong value. It's always going to be the length of the for loop above it (Line 150), rather than by the forEach that this line is nested in (Line 156). I think that the i referenced here is meant to be the second argument of the forEach, but I don't know enough about the code to know if I'm right or not.

Reply:
Is the index really necessary? I think it's sufficient to send id/name (create the evtDetail object in `init` once). For sound pools, this event will get fired a lot, and this will create a bunch of objects.

Actually don't think this event in general is really needed?

Reply:
Ah, that makes sense. I'll replace them with `evtDetail` then. The event seems to be wanted by the reporter. Sounds like the sort of events that would be useful to chain sounds one after the other, no?

Reply:
I think that's a rare case, and if dev wants, they can wire the events up themselves. But sound pooling is used when you have lots of sounds that fire rapidly and overlapping each other.

Dev can either reach and grab the pool, or pass in an audio element, or keep track of the sound triggers elsewhere, etc.

Reply:
Do you want a new PR that removes the event then?

Reply:
It's probably okay right now. At the moment, I've been making my own sound pools not using the sound component. It's fine for now